python/sdist/custom_commands.py

```diff
@@ -86,8 +86,24 @@ def build_libraries(self, libraries):
         import distutils.ccompiler
         distutils.ccompiler.CCompiler.compile = compile_parallel
 
+        # start new code
+        compilerType = self.compiler.compiler_type
```

Comment:
Move this block to free function. Use snake_case.

Reply:
I don't understand. What is free function, what is snake-case?

Reply:
a free function is a function not part of any class to similar. just a `def ....` at module level. that makes things more modular/readable/reusable. 

snake-case: `compilerType`-> `compiler_type`

Reply:
Ok, so the block starting with `# start new code` and ending with `# end new code` (currently lines 89 and 96) should be moved out of `build_libraries` and out of all classes, putting it e.g. at line 18. Should it then be called in line 89?
And then I see that you prefer snake_case to camelCase in Python, but I intentionally avoided naming the variable the same as the existing attribute, so I could call it `comp_type`. Is that OK?

Reply:
no problem to use the same name. the class attribute is only accessible via `self`, so there is no shadowing.

Reply:
OK, I'll use the same name. How about the question of where it should go? Move it to line 18 and call it in line 89?

Reply:
done